File: Data Structures & Algorithms/search-insert-position/submission-0.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int l = 0;
        int r = nums.size()-1;
        if(target > nums[r]) return nums.size();
        if(target < nums[l]) return 0;

        while (l<= r)
        {
            int m = (l + r) / 2;
            if (nums[m] < target)
            {
                l = m +1;
            }
            else if (nums[m] > target)
            {
                r = m -1;
            }
            else return m;
        }

        if (target > nums[l] && target < nums[r])
        {
            return l;
        }
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cs
public class Solution {
    public int LongestConsecutive(int[] nums) {
        var hashSet = new HashSet<int>(nums);
        var longest = 0;
        foreach (var num in hashSet)
        {
            if(!hashSet.Contains(num-1))
            {
                var length = 1;
                while(hashSet.Contains(num+length))
                {
                    length++;
                }
                longest = Math.Max(longest, length);
            }
        }

        return longest;
    }
}
